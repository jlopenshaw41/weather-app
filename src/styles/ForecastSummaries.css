.forecast-summaries {
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
    width: 50%;
}


@media only screen and (max-width: 750px) {
    .forecast-summaries {
        width: 80%;
        margin-bottom: 20px;
    }
}