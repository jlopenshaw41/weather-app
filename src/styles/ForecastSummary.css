.forecast-summary {
    border-radius: 25px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    padding: 10px 5px;
    background-color: rgba(255,255,255,0.7);
    font-size: 0.8rem;
    width: 16%;
    text-align: center;
}

.forecast-summary > button {
    border: none;
    font-size: 0.75rem;
    padding: 0px;
    margin: 5px;
    background-color: rgba(255,255,255, 0);
    font-family: "Montserrat", sans-serif;
    outline: none;
    cursor: pointer;
}

.forecast-summary > div {
    margin: 3px;
}

.forecast-summary__icon {
    font-size: 35px;
    padding: 8px;
}

@media only screen and (max-width: 750px) {
    .forecast-summary {
        width: 16%;
        padding: 5px 4px;
    }
    
    .forecast-summary > button {
        font-size: 0.5rem;
        padding: 0px;
        margin: 5px;
    }
    
    .forecast-summary > div {
        margin: 3px;
        padding: 0px;
    }
    
    .forecast-summary__icon {
        font-size: 20px;
        padding: 5px;
    }
}