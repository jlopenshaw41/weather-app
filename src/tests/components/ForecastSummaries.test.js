import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummaries", () => {
  const validProps = [
    {
      date: 1111111,
      description: "Stub description 1",
      icon: "StubIcon1",
      temperature: {
        max: 22,
        min: 12,
      },
    },
    {
      date: 2222222,
      description: "Stub description 2",
      icon: "StubIcon2",
      temperature: {
        max: 23,
        min: 13,
      },
    },
    {
      date: 3333333,
      description: "Stub description 3",
      icon: "StubIcon3",
      temperature: {
        max: 24,
        min: 14,
      },
    },
  ];
  it("renders as expected", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  xit("renders the correct number of instances of ForecastSummary", () => {});
});
