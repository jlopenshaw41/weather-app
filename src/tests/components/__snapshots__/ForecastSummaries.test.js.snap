// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders as expected 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        1111111
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        StubIcon1
      </div>
      <div
        class="forecast-summary__temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 1
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        2222222
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        StubIcon2
      </div>
      <div
        class="forecast-summary__temperature"
      >
        23°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 2
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        3333333
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        StubIcon3
      </div>
      <div
        class="forecast-summary__temperature"
      >
        24°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 3
      </div>
    </div>
  </div>
</DocumentFragment>
`;
